app/components/listing/ResourceInfos.jsx

```diff
@@ -131,96 +104,30 @@ PhoneNumber.propTypes = {
 };
 
 
-function ExternalLink(props) {
-  return <a href={props.to} target="_blank" rel="noopener noreferrer">{props.children}</a>;
-}
+const ExternalLink = ({ children, to }) => <a href={to} target="_blank" rel="noopener noreferrer">{children}</a>;
 
 ExternalLink.propTypes = {
   to: PropTypes.string.isRequired,
 };
 
 
-function Website(props) {
-  if (!props.website) {
-    return null;
-  }
-  return (
-    <span className="website">
-      <ExternalLink to={props.website}>{props.website}</ExternalLink>
-    </span>
-  );
-}
+export const Website = ({ website }) => (
+  <span className="website">
+    <ExternalLink to={website}>{website}</ExternalLink>
+  </span>
+);
 
 Website.propTypes = {
-  website: PropTypes.string,
+  website: PropTypes.string.isRequired,
 };
 
-Website.defaultProps = {
-  website: null,
-};
 
-
-function Email(props) {
-  if (!props.email) {
-    return null;
-  }
-  return (
-    <span className="email">
-      <ExternalLink to={`mailto:${props.email}`}>{props.email}</ExternalLink>
-    </span>
-  );
-}
+export const Email = ({ email }) => (
```

Comment:
Why is it preferable to export these components with their declaration? Is there a linter complaint / style reason why we don't want the export block at the bottom of the file? Personally I thought it made it clearer which components this file offers.

Reply:
There isn't a linter rule that enforces this; my personal style preference was to put the export declarations in the definitions of the components, since it makes it easier to tell whether or not a component is being exported while you're reading the definition of the component itself. However, the tradeoff is that you don't have a single place that lists all of the exported components.

I don't have a strong opinion either way, and I think this particular file is a bit of an exception because it defines so many components, some of which are exported and some of which are not, so I can revert back to the old style of declaring the exports at the bottom.